### PR TITLE
convert nspawn syscall blacklist into a whitelist (and related stuff)

### PR DESCRIPTION
**Pull request by poettering**
Let's beef up nspawn's seccomp logic a bit, and lock things down further.

### CODE REVIEW

**Comment by poettering** [comment]
@alban @robertgzr @iaguis maybe something for you to have a look at/test with rkt?

File: src/shared/seccomp-util.c [hunk 1]
```diff
@@ -689,13 +725,51 @@ const SyscallFilterSet syscall_filter_sets[_SYSCALL_FILTER_SET_MAX] = {
                 "setuid\0"
                 "setuid32\0"
         },
+        [SYSCALL_FILTER_SET_SIGNAL] = {
+                .name = "@signal",
+                .help = "Process signal operations",
+                .value =
+                "kill\0"
```
Comment (keszybz) [inline src/shared/seccomp-util.c]: Hmm, `kill` is the only one that is about sending signals, rather then receiving. The .help string also does not apply to `kill`. So maybe it should be excluded from this list, to make the list "safe".

File: man/systemd-run.xml [hunk 2]
```diff
@@ -239,7 +239,7 @@
         <command>systemd-run</command> command itself. This allows <command>systemd-run</command>
         to be used within shell pipelines.
         Note that this mode is not suitable for interactive command shells and similar, as the
-        service process will become a TTY controller when invoked on a terminal. Use <option>--pty</option> instead
+        service process will not become a TTY controller when invoked on a terminal. Use <option>--pty</option> instead
```
Comment (keszybz) [inline man/systemd-run.xml:242]: Yikes, sorry.

File: src/shared/seccomp-util.c (hunk 1 above)
Comment (poettering) [reply src/shared/seccomp-util.c]: indeed. Dropped it here now. kill is part of `@process` anyway, and i figure we don't need it in both. Will force push a new version shortly.

**Comment by iaguis** [comment]
> @alban @robertgzr @iaguis maybe something for you to have a look at/test with rkt?

Can we wait until next week so @robertgzr can give it a try?

**Comment by keszybz** [comment]
Nah, we don't want to wait that long. But it'd certainly be great if @robertgzr could test master branch after it is merged.

**Comment by keszybz** [comment]
Tests are timing out today, let's merge without waiting.

**Comment by tixxdz** [comment]
@poettering sorry I could not comment on this one... I opened this issue:
https://github.com/systemd/systemd/issues/6955

I think that the `@credentials` group should be removed, and most of the syscalls there should be allowed by default.

P.S. I missed the PR, the name was not that abvious...

**Pull request merged**